feat: Updated FAQ documentation (#50)
godot-escoria/escoria-docs: advanced/faq.rst

   :newgame
   change_scene res://game/rooms/startroom/startroom.tscn

How do I change the characters costume?
---------------------------------------

There are two ways to change the look of a character. The first is to create

> an entirely new character (i.e. a new Godot Scene with an ESCPlayer node,

Reviewer: ```suggestion
an entirely new character (i.e. a new Godot Scene with an `ESCPlayer` node,
```

> animated sprite, collision area, etc). The second option is to change just the

Reviewer: ```suggestion
animated sprite, collision area, etc). The second option is to change the
```

animations (i.e. the sprites used) for the character.

Changing the entire player scene
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> This option might be appropriate if you needed to change something fundamental

Reviewer: ```suggestion
This option might be appropriate if you need to change something fundamental
```

about a character for a particular scene (e.g. if for a specific level you only
want the character to be able to walk in 2 directions where they normally have
8 directions defined). If you choose this option, create the character scene,

> then in your game room (ESCRoom) set the PlayerScene parameter to point at the

Reviewer: ```suggestion
then in your game room (`ESCRoom`) set the `Player Scene` parameter to point at the
```

new character scene.

> Changing the character animations

Reviewer: ```suggestion
Changing the character animations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Author: @BHSDuncan  I'll add this separately. Git won't let me accept this suggestion for unknown reasons.

Changing the sprite set of a character so they look different (adding a hat,

> glasses, or changing their clothes for example) is an easy task. The process

Reviewer: ```suggestion
glasses, or changing their clothes, for example) is an easy task. The process
```

involves creating new animations, then telling Escoria to change the character

> to use the animations as the default at the appropriate time.

Reviewer: ```suggestion
to use the animations as its default at the appropriate time.
```

Create new animations
^^^^^^^^^^^^^^^^^^^^^

> You should already have idle, talk and walk animations defined for your

Reviewer: ```suggestion
You should already have idle, talk, and walk animations defined for your
```

character. Open your character's animated sprite.

.. image:: img/character_animated_sprite_node.png
   :alt: Animated sprite node

Looking at the Animations window you should see the SpriteFrames defined.

.. image:: img/character_animations_original.png
   :alt: Defined animations

Create new animations matching your new spriteset and call them something

> appropriate. In this example the demonstration character has had put on some
> jester clothes.

Reviewer: ```suggestion
appropriate. In this example, the demonstration character has had some
jester clothes put on.
```

.. image:: img/character_animations_additional.png
   :alt: Updated animations

> Go back to your character (ESCPlayer) node and using the dropdown arrow on the

Reviewer: ```suggestion
Go back to your character (`ESCPlayer`) node and, using the dropdown arrow on the
```

Animations parameter, select the menu option to create a new
**ESCAnimationResource**.

.. image:: img/character_animations_tres.png
   :alt: Updated animations

As when you created your player originally, set up the correct number of
directions for the character with the associated direction angles, as well as
the direction, idles and speak animations pointing at your newly created
animations.

.. image:: img/character_animations_tres2.png
   :alt: Updated animations

> Use the dropdown again choosing the **save** option. Give the file

Reviewer: ```suggestion
Use the dropdown again, choosing the **save** option. Give the file
```

> an appropriate name and location in the file dialogue window that appears.

Reviewer: ```suggestion
an appropriate name and location in the file dialog window that appears.
```

Assign the new animations to the character
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Now that you have the animations defined, you need to tell Escoria when you

> want to use them. If the player buys a Jester outfit for example, you may

Reviewer: ```suggestion
want to use them. If the player buys a Jester outfit, for example, you may
```

script the "use" option for the Jester outfit inventory item to change the
animation set. The command used is "set_animations", and you pass it the
path to your **ESCAnimationResource** file.

inventory_jester_outfit.esc::
  :use

  set_animations player res://game/characters/mark/mark_animations_jester.tres

How do I translate my game into other languages?
------------------------------------------------

Escoria takes advantage of Godot's built-in translation functionality for

> providing language support. Translations information is found in Godot's
> Project/Project Settings/Localization menu (text in Translations, audio in
> Remaps).

Reviewer: ```suggestion
providing language support. Translation information is found in Godot's
`Project/Project Settings/Localization` menu (text in `Translations`, audio in
`Remaps`).
```
